mongo/mongocryptd.go

```diff
@@ -77,6 +77,10 @@ func newMcryptClient(opts *options.AutoEncryptionOptions) (*mcryptClient, error)
 
 // markCommand executes the given command on mongocryptd.
 func (mc *mcryptClient) markCommand(ctx context.Context, dbName string, cmd bsoncore.Document) (bsoncore.Document, error) {
+	// Remove the explicit session from the context if one is set.
+	// The explicit session will be from a different client.
+	// If an explicit session is set, it is applied after automatic encryption.
+	ctx = NewSessionContext(ctx, nil)
```

Comment:
GODRIVER-2147 describes the error occurring in automatic decryption. The user configured their client with `SetBypassAutoEncryption(true)`. 

The `mcryptClient.client` was never the same as the parent `Client`.

Prior to the regression introduced, passing an explicit session to an operation undergoing automatic encryption resulted in a similar error.